src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/IISUrlRewriteRule.cs

```diff
@@ -57,6 +61,32 @@ public virtual void ApplyRule(RewriteContext context)
                 }
             }
 
+            if (ServerVariables != null)
+            {
+                foreach (ServerVariable serverVariable in ServerVariables)
+                {
+                    var name = serverVariable.Name;
+                    var value = serverVariable.Evaluate(context, initMatchResults, condMatchRes);
+
+                    IHeaderDictionary headerDictionary;
+                    switch (serverVariable.Type)
+                    {
+                        case ServerVariableType.Request:
+                            headerDictionary = context.HttpContext.Request.Headers;
+                            context.Logger?.RequestHeaderAdded(name, value);
+                            break;
+                        case ServerVariableType.Response:
+                            headerDictionary = context.HttpContext.Response.Headers;
+                            context.Logger?.ResponseHeaderAdded(name, value);
+                            break;
+                        default:
+                            throw new ArgumentOutOfRangeException($"Unexpected server variable: `{serverVariable.Type}`");
+                    }
+
+                    headerDictionary.Append(name, value);
```

Comment:
@mikaelm12 this is another behavior we need to verify against the naitive module, What happens when the header already exists? Replace? Append?